### DISCUSSION TITLE
What file formats are supported for input?

### DISCUSSION THREAD

**Discussion by rumpleniltkins** [openai/whisper#799] [Q&A] [answered]
Just wondering what file types are supported by the model. Are ogg vorbis files acceptable input?

**Comment by glangford** [comment, marked as answer]
ffmpeg is used to load audio (see the link to the code below), so the input type must be supported by ffmpeg. Note that you can also use a container such as .mp4 of audio+video as input.

https://github.com/openai/whisper/blob/28769fcfe50755a817ab922a7bc83483159600a9/whisper/audio.py#L22

**Reply by glangford** [reply]
See `ffmpeg -formats`; on my platform there are 395 entries, I don't know if they all work :)

**Comment by gwpl** [comment]
Can we ask for support by Whisper API for OGG/OGA and FLAC? 
(sorry if I didn't find right place to file such request).

As I read on https://platform.openai.com/docs/api-reference/audio/create#audio/create-file , official API does not support currently two very nice opensource formats:

* ogg/oga https://en.wikipedia.org/wiki/Vorbis
* FLAC https://en.wikipedia.org/wiki/FLAC

I have tons of recordings and it as default on a lot of software and devices.

As we see from example snippet OGG/OGA is supported:

```
for i in *.ogg; do
  ffmpeg -acodec libvorbis -i "$i" -acodec pcm_s16le "${i%ogg}wav"
done
```
( from https://stackoverflow.com/a/62267248/ )

```
ffmpeg -i inputfile.flac output.wav
```

from https://stackoverflow.com/a/23380032/544721

Can we please add support in Whipser API for OGG/OGA and FLAC formats? 
Looking at codebase https://github.com/openai/whisper/blob/c09a7ae299c4c34c5839a76380ae407e7d785914/whisper/audio.py#L46 it looks that it's not a problem on whisper python level but more on Whisper API side.

Otherwise, anyway I would be thankful for making sure it works well with github.com/openai/whisper for local deployments.

**Reply by SwiftAdviser** [reply]
It's my pain too! Please, team, add ogg/oga files support! 🙏

**Reply by jacksbox** [reply]
+1 for `.ogg` files

**Reply by rstockm** [reply]
+1 for FLAC - it's the most popular format amongst german podcasters

**Reply by pachocastillosr** [reply]
+1 for OGG

**Reply by jongwook** [reply]
FLAC and OGG are now supported in the Whisper API!

**Reply by gwpl** [reply]
Thank you! That's a great news!

Shouldn't documentation get updated to celebrate this! https://platform.openai.com/docs/api-reference/audio/create#audio/create-file ?

**Reply by shyamal-anadkat** [reply]
we should get the docs updated!

**Reply by rattrayalex** [reply]
Proposed adding this to the OpenAPI spec here: https://github.com/openai/openai-openapi/pull/64

**Comment by Hobbymaker-de** [comment]
.opus and .amr supported, too. (used by messengers)

**Reply by SwimSwag10** [reply]
Could you provide the documentational proof that .amr and .opus are supported? Last I checked they were not (at least .amr was not).

**Reply by Hobbymaker-de** [reply]
I wrote the comment, because there is no documentation. But it is my daily business to transcript .opus and .amr files from WhatsApp-Chats and SKY-Messenger - and it works well; I transcribed thousands.....
I think it depends on the ffmpeg.
